lib/internal/per_context/primordials.js

```diff
@@ -477,6 +477,34 @@ primordials.SafePromiseAll = (promises, mapFn) =>
     SafePromise.all(arrayToSafePromiseIterable(promises, mapFn)).then(a, b)
   );
 
+/**
+ * Should only be used for internal functions, this would produce similar
+ * results as `Promise.all` but without prototype pollution, and the return
+ * value is not a genuine Array but an array-like object.
+ * @param {Promise<any>[]} promises
+ * @returns {Promise<any[]>}
+ */
+primordials.SafePromiseAllReturnArrayLike = async (promises) => {
+  const { length } = promises;
+  const returnVal = { __proto__: null, length };
```

Comment:
why does it have to be an arraylike instead of an actual array, with proto set to null? (meaning, something `Array.isArray` returns true for)